src/handlers/thirdparty.ts

```diff
@@ -49,6 +50,62 @@ export function handleThirdPartyAdded(event: ThirdPartyAdded): void {
   metric.save()
 }
 
+
+export function handleThirdPartyUpdated(event: ThirdPartyUpdated): void {
+  if (!isURNValid(event.params._thirdPartyId)) {
+    log.error('A third party was added with an invalid URN as an id "{}"', [
+      event.params._thirdPartyId
+    ])
+    return
+  }
+
+  let thirdParty = ThirdParty.load(event.params._thirdPartyId)
+
+  if (!thirdParty) {
+    log.error('Invalid third party "{}"', [
+      event.params._thirdPartyId
+    ])
+    return
+  }
+
+  thirdParty.resolver = event.params._resolver
+  thirdParty.rawMetadata = event.params._metadata
+
+  thirdParty.save()
+
+  let eventManagersAddresses = event.params._managers
+  let eventManagersValues = event.params._managerValues
+  let managers = thirdParty.managers
+
+  for (let i = 0; i < event.params._managers.length; i++) {
+    let manager = eventManagersAddresses.pop()
+    let value = eventManagersValues.pop()
+
+    if (value) {
+      managers.push(manager.toHexString())
+      thirdParty.managers = managers
+    } else {
+      let newManagers = new Array<string>()
+
+      for (let i = 0; i < managers.length; i++) {
+        if (managers![i] != manager.toHexString()) {
+          newManagers.push(managers![i])
+        }
+      }
```

Comment:
If what you want is to remove the manager whose value is not true, you can use a filter!
```suggestion
      managers = thirdParty.managers.filter((aManager) => aManager != manager.toHexString())
```
It's one line, it works because we have the latest version of the compiler and it removes the need of using indexes, specially because we already are inside a for loop with indexes.